File: app/src/main/java/com/davismiyashiro/weathermapapp/model/OpenWeatherApi.java
package com.davismiyashiro.weathermapapp.model;

import com.davismiyashiro.weathermapapp.model.data.Place;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by Davis Miyashiro.
 */

public interface OpenWeatherApi {

    //http://api.openweathermap.org/data/2.5/forecast?q=London&appid=3e29cf11d4eabe8eba6cf25d535eaac2&cnt=5

    String APP_ID = "3e29cf11d4eabe8eba6cf25d535eaac2";

    @GET("forecast")
    Observable<Place> getWeatherFromPlace (@Query("q") String place, @Query("appid") String id);

    @GET("forecast?appid="+ APP_ID)
    Observable<Place> getForecastById (@Query("id") int place);
}

File: app/src/main/java/com/davismiyashiro/weathermapapp/model/OpenWeatherApi.kt
/*
 * MIT License
 *
 * Copyright (c) 2018 Davis Miyashiro
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.davismiyashiro.weathermapapp.model

import com.davismiyashiro.weathermapapp.model.data.Place

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by Davis Miyashiro.
 */

//http://api.openweathermap.org/data/2.5/forecast?q=London&appid=3e29cf11d4eabe8eba6cf25d535eaac2&cnt=5
interface OpenWeatherApi {

    @GET("forecast")
    fun getWeatherFromPlace(@Query("q") place: String,
                            @Query("appid") id: String): Observable<Place>

    @GET("forecast")
    fun getForecastById(@Query("id") place: Int): Observable<Place>
}
